modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -175,11 +175,30 @@
 
 (defmethod driver/describe-table :bigquery-cloud-sdk
   [_ database {table-name :name, dataset-id :schema}]
-  {:schema dataset-id
-   :name   table-name
-   :fields (-> (.. (get-table database dataset-id table-name) getDefinition getSchema)
-               table-schema->metabase-field-info
-               set)})
+  (let [table                  (get-table database dataset-id table-name)
+        tabledef               (.getDefinition table)
+        ;; a table can only have one partitioned field
+        partitioned-field-name (some-> (or (.getTimePartitioning tabledef)
+                                           (.getRangePartitioning tabledef))
+                                       .getField)
+        fields                 (set
+                                (map
+                                 #(assoc % :database-partitioned (= (:name %) partitioned-field-name))
+                                 (table-schema->metabase-field-info (. tabledef getSchema))))]
+    {:schema dataset-id
+     :name   table-name
+     :fields (cond-> fields
+               ;; if table has time partition but no field is specified as partitioned
+               ;; meaning this table is partitioned by ingestion time
+               ;; so we manually sync this virtual column
```

Comment:
It's called a pseudo-column in [the docs](https://cloud.google.com/bigquery/docs/querying-partitioned-tables#better_performance_with_pseudo-columns), maybe we should be consistent
```suggestion
               ;; so we manually sync this pseudo-column
```